lib/hawkular/operations/operations_api.rb

```diff
@@ -215,7 +215,10 @@ def check_pre_conditions(hash = {}, params = [], &callback)
       fail 'callback must have the perform method defined. include Hawkular::Operations' unless
           callback.nil? || callback.respond_to?('perform')
       params.each do |property|
-        fail "Hash property #{property} must be specified" if hash[property].nil?
+        if hash[property].nil?
+          error_message = "Hash property #{property} must be specified"
+          callback.nil? ? fail(ArgumentError, error_message) : callback.perform(:failure, error_message)
```

Comment:
I might be missing something here, but here if callback is empty then it throws an ArgumentError. A few lines up on line 216, there is "fail .... **unless** callback.nil?" check that sounds a bit contradictory.